pandas/core/indexes/base.py

```diff
@@ -2536,12 +2588,36 @@ def asof(self, label):
 
     def asof_locs(self, where, mask):
         """
-        where : array of timestamps
-        mask : array of booleans where data is not NA
+        Finds the locations (indices) of the labels from the index for
+        every entry in the `where` argument.
+
+        As in the `asof` function, if the label (a particular entry in
+        `where`) is not in the index, the latest index label upto the
+        passed label is chosen and its index returned.
+
+        If all of the labels in the index are later than a label in `where`,
+        -1 is returned.
+
+        `mask` is used to ignore NA values in the index during calculation.
+
+        Parameters
+        ----------
+        where : Index
+            An Index consisting of an array of timestamps.
+        mask : array-like
+            Array of booleans denoting where values in the original
+            data are not NA.
+
+        Returns
+        -------
+        result: array
```

Comment:
```suggestion
        numpy.ndarray
```
We used to have a name for what is being returned, but unless more than one thing is returned, we just say the type now

Reply:
I see. It should be noted then that this convention is not followed everywhere in the code. I see the name still listed in many places.

Reply:
I know, there is a PR to detect those cases, and after it's merged I'll open an issue so all them are fixed.